src/gen_rules.ml

```diff
@@ -671,7 +671,11 @@ module Gen(P : Install_rules.Params) = struct
       loop stanzas [] []
     in
     Option.iter (Merlin.merge_all merlins) ~f:(fun m ->
-      Merlin.add_rules sctx ~dir:ctx_dir ~scope ~dir_kind:kind
+      let more_src_dirs =
+        List.map (Dir_contents.dirs dir_contents) ~f:(fun dc ->
+          Path.drop_optional_build_context (Dir_contents.dir dc))
```

Comment:
I find it a bit weird that we're dropping the build context for more_src_dirs but then dir still has the build prefix which Merlin.add_rules will strip itself. So I'd prefer if merlin did the of the build context to all dirs consistently.

Reply:
On the other hand this is consistent with the `source_dirs` argument of `Merlin.make`. I've been wondering if we shouldn't have different types for paths to make these things clearer.